framework/auth/views.py

```diff
@@ -378,30 +378,57 @@ def auth_register(auth):
 
     raise HTTPError(http.BAD_REQUEST)
 
-
-def auth_logout(redirect_url=None):
+@collect_auth
+def auth_logout(auth=None, redirect_url=None, next_url=None):
```

Comment:
I believe `auth` is guaranteed to not be `None` here, as `@collect_auth` will inject a user-less `Auth` object if unauthenticated.